### PR TITLE
boards, module and Makefile: drop iotools

### PR DESCRIPTION
**Pull request by macpijan**
As discussed in: linuxboot#1704
there is no need to include iotools module for nitropads.

### CODE REVIEW

**Comment by JonathonHall-Purism** [comment]
@macpijan This was the only use of iotools right?  If so please remove modules/iotools and the bits from Makefile

**Comment by macpijan** [comment]
>  If so please remove modules/iotools and the bits from Makefile

I removed it at first.

@tlaurion Said it was not the only user of it, now I am confused: https://github.com/linuxboot/heads/pull/1704#issuecomment-2182875878

Perhaps you meant I should remove it from these as well?

```
./qemu-coreboot-fbwhiptail-tpm1-hotp/qemu-coreboot-fbwhiptail-tpm1-hotp.config:#CONFIG_IOTOOLS=y
./qemu-coreboot-fbwhiptail-tpm1/qemu-coreboot-fbwhiptail-tpm1.config:CONFIG_IOTOOLS=y
./qemu-coreboot-fbwhiptail-tpm2-hotp/qemu-coreboot-fbwhiptail-tpm2-hotp.config:CONFIG_IOTOOLS=y
./qemu-coreboot-fbwhiptail-tpm2/qemu-coreboot-fbwhiptail-tpm2.config:CONFIG_IOTOOLS=y
./qemu-coreboot-whiptail-tpm1-hotp/qemu-coreboot-whiptail-tpm1-hotp.config:CONFIG_IOTOOLS=y
./qemu-coreboot-whiptail-tpm1/qemu-coreboot-whiptail-tpm1.config:CONFIG_IOTOOLS=y
./qemu-coreboot-whiptail-tpm2-hotp/qemu-coreboot-whiptail-tpm2-hotp.config:#CONFIG_IOTOOLS=y
./qemu-coreboot-whiptail-tpm2/qemu-coreboot-whiptail-tpm2.config:#CONFIG_IOTOOLS=y
./x230-hotp-maximized/x230-hotp-maximized.config:#CONFIG_IOTOOLS=y
```

?

**Comment by tlaurion** [comment]
@JonathonHall-Purism yes.

no more iotools calls in codebase outside of [initrd/bin/nitropad-shutdown.sh](https://github.com/linuxboot/heads/pull/1704/commits/fd98c8d0d9f64e8ef37e45afe9edf4c3b8e0fe8d#diff-8bf3d78c2f2d0e42afe466f6f0a70c56ce2f7548c296ad28a71c476c1ba4da65) which vanished under https://github.com/linuxboot/heads/pull/1704

**Comment by tlaurion** [comment]
> > If so please remove modules/iotools and the bits from Makefile
> 
> I removed it at first.
> 
> @tlaurion Said it was not the only user of it, now I am confused: [#1704 (comment)](https://github.com/linuxboot/heads/pull/1704#issuecomment-2182875878)
> 
> Perhaps you meant I should remove it from these as well?
> 
> ```
> ./qemu-coreboot-fbwhiptail-tpm1-hotp/qemu-coreboot-fbwhiptail-tpm1-hotp.config:#CONFIG_IOTOOLS=y
> ./qemu-coreboot-fbwhiptail-tpm1/qemu-coreboot-fbwhiptail-tpm1.config:CONFIG_IOTOOLS=y
> ./qemu-coreboot-fbwhiptail-tpm2-hotp/qemu-coreboot-fbwhiptail-tpm2-hotp.config:CONFIG_IOTOOLS=y
> ./qemu-coreboot-fbwhiptail-tpm2/qemu-coreboot-fbwhiptail-tpm2.config:CONFIG_IOTOOLS=y
> ./qemu-coreboot-whiptail-tpm1-hotp/qemu-coreboot-whiptail-tpm1-hotp.config:CONFIG_IOTOOLS=y
> ./qemu-coreboot-whiptail-tpm1/qemu-coreboot-whiptail-tpm1.config:CONFIG_IOTOOLS=y
> ./qemu-coreboot-whiptail-tpm2-hotp/qemu-coreboot-whiptail-tpm2-hotp.config:#CONFIG_IOTOOLS=y
> ./qemu-coreboot-whiptail-tpm2/qemu-coreboot-whiptail-tpm2.config:#CONFIG_IOTOOLS=y
> ./x230-hotp-maximized/x230-hotp-maximized.config:#CONFIG_IOTOOLS=y
> ```
> 
> ?

Sorry. What I meants is that boards, mostly testing boards (qemu) where referencing it (including it for testing purposes) while if the modules vanishes and boards still refering to it, they would break at builds.

TLDR: all references to IOTOOLS need to disappear, including Makefile conditional inclusion. Makes sense?

**Comment by tlaurion** [comment]
Renamed PR to match goal, will close #1706

**Review by JonathonHall-Purism — APPROVED** [review]
LGTM, thanks :bow:

**Comment by tlaurion** [comment]
@macpijan thanks for contributing upstream! Much appreciated!!!!!! Merging
